plugins/pastefromword/plugin.js

```diff
@@ -167,6 +178,40 @@
 					}
 				}
 			}
+
+			function handleBlobs( blobs, callback ) {
+				var blobUrlsToProcess = removeDuplicates( blobs ),
+					blobUrlsToBase64Map = {},
+					amountOfBlobsToProcess = blobUrlsToProcess.length;
+
+				CKEDITOR.tools.array.forEach( blobUrlsToProcess, function( blobUrl ) {
+					CKEDITOR.tools.convertBlobUrlToBase64( blobUrl, function( base64 ) {
+						blobUrlsToBase64Map[ blobUrl ] = base64;
+						amountOfBlobsToProcess--;
+						if ( amountOfBlobsToProcess === 0 ) {
+							replaceBlobUrlsInEditor( blobUrlsToBase64Map );
+							callback();
+						}
+					}, this );
+				}, this );
+
+
+				function removeDuplicates( arr ) {
+					return CKEDITOR.tools.array.filter( arr, function( item, index ) {
+						return index === CKEDITOR.tools.array.indexOf( arr, item );
+					} );
+				}
+
+				function replaceBlobUrlsInEditor( map ) {
+					for ( var blob in map ) {
+						var nodeList = editor.editable().find( 'img[src="' + blob + '"]' ).toArray();
+						CKEDITOR.tools.array.forEach( nodeList, function( element ) {
```

Comment:
Amount of `CKEDITOR.tools` in `handleBlobs` is too high. Let's make some short alias, which will increase the readability.